In [4]:
import pandas as pd
import numpy as np
import statsmodels.graphics.tsaplots as sgt
import statsmodels.tsa.stattools
import seaborn as sns

In [11]:
### prepare data
raw_csv = pd.read_csv("resources/Index2018.csv")
df_comp = raw_csv.copy()

### keep only one column
df_comp['date'] = pd.to_datetime(df_comp.date , dayfirst = True)
df_comp.set_index('date' , inplace = True)
df_comp['market_value'] = df_comp.spx
df_comp = df_comp[['market_value']]

wn = np.random.normal(loc = df_comp['market_value'].mean() , scale = df_comp.market_value.std() ,\
                      size = len(df_comp))
df_comp['wn'] = wn

rw = pd.read_csv("resources/RandWalk.csv")
rw['date'] = pd.to_datetime(rw.date , dayfirst = True)
rw.set_index('date' , inplace = True)
rw = rw.asfreq('b')

## Stationary

### For stock price:

In [9]:
statsmodels.tsa.stattools.adfuller(df_comp.market_value)

(1.0724042027036522,
 0.9949740839827675,
 18,
 6250,
 {'1%': -3.4313967180469573,
  '5%': -2.862002556554404,
  '10%': -2.5670162159104},
 50197.90170692382)

test value = 1.0724042027036522\
p value = 0.9949740839827675

**We don't have enough evidence to conclude the data in stationary**


### For White noise

However, for **white noise** data:

In [10]:
statsmodels.tsa.stattools.adfuller(df_comp.wn)

(-77.45300320987212,
 0.0,
 0,
 6268,
 {'1%': -3.4313937109235875,
  '5%': -2.862001227905827,
  '10%': -2.5670155086390123},
 94834.35691805962)

### For Random walk

In [12]:
statsmodels.tsa.stattools.adfuller(rw)

(-1.328607392768971,
 0.6159849181617387,
 24,
 4996,
 {'1%': -3.4316595802782865,
  '5%': -2.8621186927706463,
  '10%': -2.567078038881065},
 46299.333497595144)

More often than not, the covariance of two random walk intervals with identical size would be different.\
Therefore, we expect random walk to be a non-stationary process.